torch/csrc/jit/passes/prepack_folding.cpp

```diff
@@ -0,0 +1,59 @@
+#include <stack>
+
+#include <torch/csrc/jit/api/module.h>
+#include <torch/csrc/jit/passes/constant_pooling.h>
+#include <torch/csrc/jit/passes/constant_propagation.h>
+#include <torch/csrc/jit/passes/prepack_folding.h>
+
+namespace torch {
+namespace jit {
+
+// Must run this pass after constant folding.
+void FoldPrePackingOps(script::Module& m,
+    const PrePackingOpsFilterFn& is_foldable_op) {
+  // Since this pass can be called by quantization or other passes as well,
+  // we need to make sure we generate a unique "packed_weight_\d+"
+  // Thus static uid.
+  static int64_t uid = 0;
+
+  auto method = m.get_method("forward");
+  auto graph = method.graph();
+  std::stack<Block*> blocks_to_visit;
+  std::unordered_set<Node*> nodes_to_delete;
+  blocks_to_visit.push(graph->block());
+  std::string attr_name_base("packed_weight_");
+  while (!blocks_to_visit.empty()) {
+    Block* b = blocks_to_visit.top();
+    blocks_to_visit.pop();
+    for (Node* n : b->nodes()) {
+      if (is_foldable_op(n)) {
+        auto optional_outputs = runNodeIfInputsAreConstant(n);
+        if (optional_outputs) {
+          auto outputs = optional_outputs.value();
+          TORCH_CHECK(outputs.size() == 1, "Prepack ops have single output");
+          auto attr_name = attr_name_base + c10::to_string(uid++);
+          m.register_attribute(attr_name, n->output(0)->type(), outputs[0]);
```

Comment:
Can we check that this is not already registered by the user?

Reply:
Do you intend to use that check so as not do registration again? Or delete the attribute and register a new one? I think the latter probably would make sense. I will have to check API to see whats possible here.

Reply:
If it's already present, I think we should throw an exception.  I think it's fine to fail the conversion if a user happens to create an attribute called "_jit_pass_packed_weight_1", but I think we should try not to produce incorrect results even if they do.

Reply:
I can add an assert that, although I dont see that happening. But well, we dont write bugs in the code knowingly, so let me add an assert.